master/docs/developer/pull-request.rst

```diff
@@ -0,0 +1,212 @@
+Submitting Pull Requests
+========================
+
+As Buildbot it is used by software developers, it tends to receive a significant number of patches.
+The most effective way to make sure your patch gets noticed and merged is to submit it via GitHub.
+This assumes some familiarity with git, but not too much. Note that GitHub has some great `​Git guides <http://github.com/guides>`_ to get you started.
+
+Guidelines
+----------
+
+* Pull requests should be based on the latest development code, not on the most recent release.
+  That is, you should check out the `master` branch and develop on top of it.
+
+* Pull requests should include code changes, relevant documentation changes, and relevant unit tests.
+  Any patch longer than a few lines which does not have documentation or tests will probably be rejected with a request to add documentation or tests.
```

Comment:
I think I would rephrase it to not use "rejected". It can feel like the PR will be closed immediately when instead we ask for these things and the PR waits until docs and tests are updated.

Perhaps:
>   Any patch longer than a few lines which does not have documentation or tests is unlikely to be merged.